Plugins/OpenAPIGeneratorCommand/plugin.swift

```
@@ -0,0 +1,103 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import PackagePlugin
+import Foundation
+
+@main
+struct SwiftOpenAPIGeneratorPlugin {
+    func runCommand(
+        targetWorkingDirectory: Path,
+        tool: (String) throws -> PluginContext.Tool,
+        sourceFiles: FileList,
+        targetName: String
+    ) throws {
+        let inputs = try PluginUtils.validateInputs(
+            workingDirectory: targetWorkingDirectory,
+            tool: tool,
+            sourceFiles: sourceFiles,
+            targetName: targetName,
+            pluginSource: .command
+        )
+
+        let toolUrl = URL(fileURLWithPath: inputs.tool.path.string)
+        let process = Process()
+        process.executableURL = toolUrl
+        process.arguments = inputs.arguments
+        process.environment = [:]
+        try process.run()
+        process.waitUntilExit()
```

Also, we should inspect the `process.terminationStatus` and throw an error if it's not 0.

---

Right i was thinking about that ... i wonder if we can report the error too somehow ...

---

It does show, as it shows the "Swift OpenAPI Generator is running with the following configuration:" logs too.